casual/casual-caller-http-client/src/test/groovy/se/laz/casual/http/HttpClientTest.groovy

```diff
@@ -0,0 +1,157 @@
+/*
+ * Copyright (c) 2024, The casual project. All rights reserved.
+ *
+ * This software is licensed under the MIT license, https://opensource.org/licenses/MIT
+ */
+
+package se.laz.casual.http
+
+import jakarta.ws.rs.core.Application
+import org.glassfish.jersey.server.ResourceConfig
+import org.glassfish.jersey.test.JerseyTest
+import se.laz.casual.api.buffer.CasualBuffer
+import se.laz.casual.api.buffer.ServiceReturn
+import se.laz.casual.api.buffer.type.CStringBuffer
+import se.laz.casual.api.buffer.type.JsonBuffer
+import se.laz.casual.api.buffer.type.OctetBuffer
+import se.laz.casual.api.buffer.type.fielded.FieldedTypeBuffer
+import se.laz.casual.api.flags.ErrorState
+import se.laz.casual.api.flags.ServiceReturnState
+import se.laz.casual.http.test.resource.TestPath
+import se.laz.casual.http.test.resource.TestResource
+import spock.lang.Shared
+import spock.lang.Specification
+
+import java.nio.charset.StandardCharsets
+
+class HttpClientTest extends Specification
+{
+   @Shared
+   HttpClient client = HttpClient.of()
+   @Shared
+   def content = '{"msg":"bazinga!"}'
+   @Shared
+   def key = 'FLD_STRING1'
+   @Shared
+   def root = 'test'
+
+   JerseyTest resource = new JerseyTest(){
+      @Override
+      protected Application configure(){
+         return new ResourceConfig(TestResource.class)
+      }
+   }
+
+   def setup()
+   {
+      resource.setUp()
+   }
+
+   def cleanup()
+   {
+      resource.tearDown()
+   }
+
+   def 'ok'()
+   {
+      given:
+      URI uri = new URI(resource.getBaseUri().toString() + "${root}/${TestPath.OK.path}")
+      ServiceReturn<CasualBuffer> response = client.request(uri, buffer)
+      expect:
+      response.getServiceReturnState() == returnState
+      response.getErrorState() == errorState
+      response.getReplyBuffer().getBytes() == buffer.getBytes()
+      where:
+      buffer                                                      || returnState                   || errorState
+      JsonBuffer.of(content)                                      || ServiceReturnState.TPSUCCESS  || ErrorState.OK
```

Comment:
no need for `returnState` or `errorState` to be in the where table as they are fixed.
Same for all other tests below.